oozie-to-airflow/mappers/git_mapper.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Maps Shell action into Airflow's DAG"""
+import shlex
+from typing import Dict, Set, Optional
+
+import xml.etree.ElementTree as ET
+from urllib.parse import urlparse
+
+from airflow.utils.trigger_rule import TriggerRule
+
+from converter.primitives import Task, Relation
+from mappers.action_mapper import ActionMapper
+from mappers.prepare_mixin import PrepareMixin
+
+from utils.template_utils import render_template
+from utils.xml_utils import get_text_or_default
+
+TAG_GIT_URI = "git-uri"
+TAG_BRANCH = "branch"
+TAG_KEY_PATH = "key-path"
+TAG_DESTINATION_URI = "destination-uri"
+
+
+def prepare_git_command(
+    git_uri: str, git_branch: Optional[str], destination_path: str, key_path: Optional[str]
+):
+    cmd = (
+        f"$DAGS_FOLDER/../data/git.sh "
+        "--cluster {dataproc_cluster} "
+        "--region {gcp_region} "
+        f"--git-uri {shlex.quote(git_uri)} "
+        f"--destination-path {destination_path}"
+    )
+
+    if git_branch:
+        cmd += f" --branch {shlex.quote(git_branch)}"
+
+    if key_path:
+        cmd += f" --key-path {shlex.quote(key_path)}"
+
+    return cmd
+
+
+class GitMapper(ActionMapper, PrepareMixin):
+    """
+    Converts a Shell Oozie action to an Airflow task.
+    """
+
+    bash_command: str
+
+    def __init__(
+        self,
+        oozie_node: ET.Element,
+        name: str,
+        trigger_rule: str = TriggerRule.ALL_SUCCESS,
+        params: Dict[str, str] = None,
+        **kwargs,
+    ):
+        ActionMapper.__init__(self, oozie_node, name, trigger_rule, **kwargs)
+        if params is None:
+            params = {}
+        self.params = params
+        self.trigger_rule = trigger_rule
+
+    def on_parse_node(self):
+        git_uri = get_text_or_default(self.oozie_node, TAG_GIT_URI, self.params)
+        git_branch = get_text_or_default(self.oozie_node, TAG_BRANCH, self.params)
+        destination_uri = get_text_or_default(self.oozie_node, TAG_DESTINATION_URI, self.params)
+        destination_path = urlparse(destination_uri).path
+        key_path_uri = get_text_or_default(self.oozie_node, TAG_KEY_PATH, self.params)
+        key_path = urlparse(key_path_uri).path
+        self.bash_command = prepare_git_command(
+            git_uri=git_uri, git_branch=git_branch, destination_path=destination_path, key_path=key_path
+        )
+
+    def convert_to_text(self):
```

Comment:
It should probably be updated to match changes introduced in: https://github.com/GoogleCloudPlatform/cloud-composer/pull/167

Reply:
I rebased the PR.